sdk/storage/azure_storage_blob/src/clients/blob_client.rs

```diff
@@ -356,8 +357,18 @@ impl BlobClient {
         self.client.get_account_info(options).await
     }
 
-    /// Returns `true` if the blob exists, and returns `false` otherwise.
-    pub async fn exists(&self) -> bool {
-        self.get_properties(None).await.is_ok()
+    /// Returns `true` if the blob exists, `false` if the blob does not exist, and propagates all other errors.
+    pub async fn exists(&self) -> Result<bool> {
+        match self.client.get_properties(None).await {
+            Ok(_) => Ok(true),
+            Err(e) if e.http_status() == Some(StatusCode::NotFound) => match e.kind() {
+                ErrorKind::HttpResponse {
+                    error_code: Some(error_code),
+                    ..
+                } if error_code == "BlobNotFound" || error_code == "ContainerNotFound" => Ok(false),
+                _ => Ok(false),
```

Comment:
You're returning false in both cases. What's the point of checking the inner error code?

Reply:
Same below.